src/libraries/System.ComponentModel.Annotations/src/System/ComponentModel/DataAnnotations/ValidationAttributeStore.cs

```diff
@@ -217,21 +217,46 @@ internal bool TryGetPropertyStoreItem(string propertyName, [NotNullWhen(true)] o
 
             private Dictionary<string, PropertyStoreItem> CreatePropertyStoreItems()
             {
-                var propertyStoreItems = new Dictionary<string, PropertyStoreItem>();
-
-                // exclude index properties to match old TypeDescriptor functionality
-                var properties = _type.GetRuntimeProperties()
-                    .Where(prop => IsPublic(prop) && !prop.GetIndexParameters().Any());
-                foreach (PropertyInfo property in properties)
+                Dictionary<string, PropertyStoreItem> propertyStoreItems = new Dictionary<string, PropertyStoreItem>();
+                PropertyDescriptorCollection properties = TypeDescriptor.GetProperties(this._type);
+                foreach (PropertyDescriptor property in properties)
                 {
-                    // use CustomAttributeExtensions.GetCustomAttributes() to get inherited attributes as well as direct ones
-                    var item = new PropertyStoreItem(property.PropertyType,
-                        CustomAttributeExtensions.GetCustomAttributes(property, true));
+                    PropertyStoreItem item = new PropertyStoreItem(property.PropertyType, GetExplicitAttributes(property).Cast<Attribute>());
                     propertyStoreItems[property.Name] = item;
                 }
 
                 return propertyStoreItems;
             }
+
+            /// <summary>
+            /// Method to extract only the explicitly specified attributes from a <see cref="PropertyDescriptor"/>
+            /// </summary>
+            /// <remarks>
+            /// Normal TypeDescriptor semantics are to inherit the attributes of a property's type.  This method
+            /// exists to suppress those inherited attributes.
+            /// </remarks>
+            /// <param name="propertyDescriptor">The property descriptor whose attributes are needed.</param>
+            /// <returns>A new <see cref="AttributeCollection"/> stripped of any attributes from the property's type.</returns>
+            public static AttributeCollection GetExplicitAttributes(PropertyDescriptor propertyDescriptor)
```

Comment:
Why is this method public? It seems to only be used by this class.

Reply:
🤔 that would be a copy & paste fail from .NET Framework. Changed to internal. Thanks

Reply:
(nit) Does it even need to be `internal`? Why not `private`?